### PR TITLE
[FEAT] Goal Command AcceptanceTest

### PR DESCRIPTION
**Pull request by sueelly**
## #️⃣ Related Issue

Linked Feature backlog: Related to [issue#3](https://github.com/dnd-side-project/dnd-12th-2-backend/issues/3)

## 📝 Purpose of PR

Goal의 Command에 대한 인수 테스트를 진행한다.

### 인수테스트
인수 테스트는 RestAssured를 사용하여 실제 웹환경처럼 작동하게끔 한다.
이때, context 생성 비용을 아끼기 위해 가급적 같은 환경끼리 묶어서 돌아가게끔 한다. -> AcceptanceTest, IntegrationTest abstract class
swagger, api docs 대신 이것을 전달한다. 이를 위해 모든 예외 처리를 포함하여 더 꼼꼼하게 작성하였다.
현재 로직은 구현되지 않아 당연히 실패하는 게 맞는 테스트이다.

### command 분리 이유
- 시각화 (조회. query)의 경우 요구사항이 명확히 정해지지도 않았을 뿐더러, 
- query의 중요도 및 복잡도가 올라감에 따라 CQS pattern, 후에는 CQRS에 대한 확장까지 고려중이기 떄문이다.

### Issue 1. Test Container
원래는 AcceptanceTest의 경우 test container를 사용하여 production 환경과 최대한 유사하게 구성하려고 했으나, container까지는 띄워지나 connection이 되지 않았다.
당장 여기에 더이상 신경쓸 수 없어 넘어갔다.

### Issue 2. Authorization
인수 테스트는 실제 유저 시나리오를 구현해야하기 때문에 authorization도 함께 검사된다.
단, 이 경우 token값을 어떻게 해야할지 생각을 해봐야할 것 같다.
당장에는 테스트용으로 기한이 긴 토큰 한 쌍을 만들고, 테스트끼리의 순서를 조정하여 앞에서 로그인 -> 토큰 발급 후 해당 토큰을 사용하면 되지 않을까 싶다.

## Checklist

-   [x] Does commit messages follow the convention?
-   [x] Are variable names brief but descriptive?
-   [x] Is the code clean and easy to understand?
-   [x] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [x] Does the test successfully test edge and corner cases?

## 💬 Review Requirements (Optional)

Is there anything you can praise about this PR? Start with praise.
Make a review, leaving kind comments and suggesting changes where needed (to resolve the above).

### CODE REVIEW

File: src/main/java/ac/dnd/dodal/domain/goal/model/Goal.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import ac.dnd.dodal.common.model.BaseEntity;
+import ac.dnd.dodal.common.exception.BadRequestException;
+import ac.dnd.dodal.common.exception.UnauthorizedException;
+import ac.dnd.dodal.common.exception.ForbiddenException;
+import ac.dnd.dodal.domain.goal.constraint.GoalConstraint;
+import ac.dnd.dodal.domain.goal.exception.GoalExceptionCode;
+
+@Entity
+@Getter
+@ToString(callSuper = true)
```
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/domain/goal/model/Goal.java:23]: toString을 사용하는 위치가 있어서 사용하신건가요? 아니면 추후 사용을 대비해서 넣어둔 설정일까요?

**Review by 5hseok — APPROVED** [review]
